config/src/logger.rs

```diff
@@ -40,6 +35,68 @@ pub enum Format {
     Json,
 }
 
+/// List of directives
+#[derive(Clone, DeserializeFromStr, SerializeDisplay, PartialEq, Eq)]
+pub struct Directives(Vec<Directive>);
+
+impl FromStr for Directives {
+    type Err = tracing_subscriber::filter::ParseError;
+
+    fn from_str(dirs: &str) -> std::result::Result<Self, Self::Err> {
+        if dirs.is_empty() {
+            return Ok(Self(Vec::new()));
+        }
+        let directives = dirs
+            .split(',')
+            .filter(|s| !s.is_empty())
+            .map(Directive::from_str)
+            .collect::<std::result::Result<Vec<_>, _>>()?;
+        Ok(Self(directives))
+    }
+}
+
+impl Display for Directives {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let mut directives_iter = self.0.iter();
+        if let Some(directive) = directives_iter.next() {
+            write!(f, "{directive}")?;
+        }
+        for directive in directives_iter {
+            write!(f, ",{directive}")?;
+        }
+        Ok(())
+    }
+}
+
+impl Debug for Directives {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        Display::fmt(&self, f)
```

Comment:
`tracing`'s `Debug` implementation for `Directive` is actually done with `#[derive(Debug)]` (looks like `Directive { in_span: None, fields: [], target: None, level: LevelFilter::INFO }`), should we follow them maybe?

Or do we want to pretty-print it as part of some structure?

Reply:
i've used pretty printing for debug, because it it used in one of our configuration tests